x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlTranslatorHandler.java

```diff
@@ -245,4 +250,31 @@ private static boolean isInRange(DataType numericFieldDataType, DataType valueDa
             return minValue.compareTo(decimalValue) <= 0 && maxValue.compareTo(decimalValue) >= 0;
         }
     }
+
+    public static class Scalars extends ExpressionTranslator<ScalarFunction> {
+        @Override
+        protected Query asQuery(ScalarFunction f, TranslatorHandler handler) {
+            return doTranslate(f, handler);
+        }
+
+        public static Query doTranslate(ScalarFunction f, TranslatorHandler handler) {
+            if (f instanceof CIDRMatch cm) {
+                if (cm.ipField() instanceof FieldAttribute && Expressions.foldable(cm.matches())) {
+                    String targetFieldName = handler.nameOf(((FieldAttribute) cm.ipField()).exactAttribute());
+
+                    Set<Object> set = Sets.newLinkedHashSetWithExpectedSize(CollectionUtils.mapSize(cm.matches().size()));
+
+                    for (Expression e : cm.matches()) {
+                        set.add(e.fold());
+                    }
```

Comment:
This can be further improved:
```
if (cm.ipField() instanceof FieldAttribute fa ...) {
     String fieldName = handler.nameOf(fa).exactAttribute()
     Set<Object> set = new LinkedHashSet<>(Expressions.fold(cm.matches());

}
```